models/gif_cond_exp_multisynapse.cpp

```diff
@@ -0,0 +1,692 @@
+  if ( B_.e_ )
+    gsl_odeiv_evolve_free( B_.e_ );
+}
+
+/* ----------------------------------------------------------------
+ * Node initialization functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::gif_cond_exp_multisynapse::init_state_( const Node& proto )
+{
+  const gif_cond_exp_multisynapse& pr =
+    downcast< gif_cond_exp_multisynapse >( proto );
+  S_ = pr.S_;
+}
+
+void
+nest::gif_cond_exp_multisynapse::init_buffers_()
+{
+  B_.spike_exc_.clear(); // includes resize
+  B_.spike_inh_.clear(); // includes resize
+  B_.currents_.clear();  //!< includes resize
+  B_.logger_.reset();    //!< includes resize
+  Archiving_Node::clear_history();
+
+  int state_size = 1 + ( State_::STATE_VEC_SIZE - 1 ) * P_.num_of_receptors_;
+
+  B_.step_ = Time::get_resolution().get_ms();
+  B_.IntegrationStep_ = B_.step_;
+
+  if ( B_.s_ == 0 )
+    B_.s_ = gsl_odeiv_step_alloc( gsl_odeiv_step_rkf45, state_size );
+  else
+    gsl_odeiv_step_reset( B_.s_ );
+
+  if ( B_.c_ == 0 )
+    B_.c_ = gsl_odeiv_control_y_new( 1e-3, 0.0 );
+  else
+    gsl_odeiv_control_init( B_.c_, 1e-3, 0.0, 1.0, 0.0 );
+
+
+  if ( B_.e_ == 0 )
+    B_.e_ = gsl_odeiv_evolve_alloc( state_size );
+  else
+    gsl_odeiv_evolve_reset( B_.e_ );
+
+  B_.sys_.function = gif_cond_exp_multisynapse_dynamics;
+  B_.sys_.jacobian = NULL;
+  B_.sys_.dimension = state_size;
+  B_.sys_.params = reinterpret_cast< void* >( this );
+}
+
+void
+nest::gif_cond_exp_multisynapse::calibrate()
+{
+
+  P_.receptor_types_.resize( P_.num_of_receptors_ );
+  for ( size_t i = 0; i < P_.num_of_receptors_; i++ )
+  {
+    P_.receptor_types_[ i ] = i + 1;
+  }
+
+  int state_size = 1 + ( State_::STATE_VEC_SIZE - 1 ) * P_.num_of_receptors_;
+
+  B_.spike_exc_.resize( P_.num_of_receptors_ );
+  B_.spike_inh_.resize( P_.num_of_receptors_ );
+  S_.y_.resize( state_size );
+
+  B_.sys_.dimension = state_size;
+
+  B_.logger_.init();
+
+  double_t h = Time::get_resolution().get_ms();
+  V_.rng_ = kernel().rng_manager.get_rng( get_thread() );
+
+  V_.RefractoryCounts_ = Time( Time::ms( P_.t_ref_ ) ).get_steps();
+  assert( V_.RefractoryCounts_
+    >= 0 ); // since t_ref_ >= 0, this can only fail in error
+
+  // initializing internal state
+  if ( !S_.sfa_stc_initialized_ )
+  {
+    V_.P_sfa_.clear();
+    V_.P_stc_.clear();
+    S_.sfa_elems_.clear();
+    S_.stc_elems_.clear();
+
+    for ( size_t i = 0; i < P_.tau_sfa_.size(); i++ )
+    {
+      V_.P_sfa_.push_back( std::exp( -h / P_.tau_sfa_[ i ] ) );
+      S_.sfa_elems_.push_back( 0.0 );
+    }
+
+    for ( size_t i = 0; i < P_.tau_stc_.size(); i++ )
+    {
+      V_.P_stc_.push_back( std::exp( -h / P_.tau_stc_[ i ] ) );
+      S_.stc_elems_.push_back( 0.0 );
+    }
+
+    S_.sfa_stc_initialized_ = true;
+  }
+}
+
+/* ----------------------------------------------------------------
+ * Update and spike handling functions
+ */
+
+void
+nest::gif_cond_exp_multisynapse::update( Time const& origin,
```

Comment:
See discussion of `update()` method in `gif_psc_exp`, it applies here correspondingly.